driver-dse/src/main/java/com/datastax/driver/auth/DseAuthProvider.java

```diff
@@ -65,20 +91,153 @@
  *       principal="user@MYDOMAIN.COM";
  * };
  * </pre>
- * <p/>
- * To connect to clusters using internal authentication, use the standard method for
- * setting credentials. eg:
- * <p/>
- * <pre>
- * Cluster cluster = Cluster.builder()
- *                          .addContactPoint(hostname)
- *                          .withCredentials("username", "password")
- *                          .build();
- * </pre>
  */
 public class DseAuthProvider implements AuthProvider {
-    @Override
+    private static final String DSE_AUTHENTICATOR = "com.datastax.bdp.cassandra.auth.DseAuthenticator";
+    private final String username;
+    private final String password;
+
+    public DseAuthProvider() {
```

Comment:
I like your second idea much bettter. That would make it very clear what was being selected. If it's ok I'd prefer ```DsePlainTextAuthProvider``` and ```DseGSSAPIAuthProvider``` but if you'd prefer your versions I'll stick with them.